lib/spack/spack/test/llnl/util/file_list.py

```diff
@@ -52,6 +65,15 @@ def header_list():
     return h
 
 
+plat_static_ext = "lib" if is_windows else "a"
+
+
+plat_shared_ext = "dll" if is_windows else ".so"
```

Comment:
@jpopelar This line is causing (at least some of) your current CI failures. The extra `.` character in the `".so"` designation for `plat_shared_ext` is causing the string comp for the library names to fail.

Reply:
This fix is already present in the RPath PR, #31930 btw. A lot of the changes from that PR seem to be present in this one, so you may want to check out that PR for future issues.